### ISSUE TITLE
Java: semgrep does not match the 'try-with-resources' pattern.

### ISSUE THREAD

**Issue by minusworld** [semgrep/semgrep#717]
**Describe the bug**
The 'try-with-resources' pattern was added in Java 8: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

This example I was using as a test case uses this pattern, but semgrep doesn't pick it up:
https://find-sec-bugs.github.io/bugs.htm#OBJECT_DESERIALIZATION

**To Reproduce**
This pattern should match both cases:
https://semgrep.live/GD1

It looks like it just gets skipped in the AST altogether:
![image](https://user-images.githubusercontent.com/2374948/81352378-3f8ffe80-907b-11ea-94ec-de74d78a0e98.png)

**Comment by aryx** [comment]
I've improved ast_java.ml to support those try with resources, but I've not fixed the grammar to generate those
resources. Also I'm not sure we are matching correctly those in semgrep.

**Comment by aryx** [comment]
@IagoAbal this is a good candidate.

**Comment by jeremybrooks-sf** [comment]
Is there any plan to merge this into the main semgrep repo from @aryx's fork? This is causing false negatives for our team.

**Comment by aryx** [comment]
This was merged, a long time ago.

**Comment by aryx** [comment]
If you have examples that do not work, feel free to create a new issue with a link to a semgrep.dev playground example showing the current limitations.

**Comment by jeremybrooks-sf** [comment]
I can create a new issue, but I'm not sure if this is a bug or just my misunderstanding. I tried the example which is posted in the original comment (similar case here https://semgrep.dev/s/zJKW). The "object deserialization" rule in that playground link does not flag line 21 as I would expect. I am assuming semgrep.live is using the latest engine. I also tried the sample code and rule from that link locally with semgrep 0.80.0 (latest release as of today) and it also does not flag. Looking at the output from --dump-ast, it does appear that the ObjectInputStream in the try-with-resources block is being parsed. I was expecting the rule to flag lines 14 and 21, but maybe the rule needs to be tweaked?
